commit/solvers.py

```diff
@@ -284,18 +284,19 @@ def fista( y, A, At, tol_fun, tol_x, max_iter, verbose, x0, omega, proximal) :
     """
 
     # Initialization
-    res = -y.copy()
+    
     xhat = x0.copy()
     x = np.zeros_like(xhat)
-    res += A.dot(xhat)
+    res = confidence_array * (A.dot(xhat) - y.copy()) 
+    
     proximal( xhat )
     reg_term = omega( xhat )
     prev_obj = 0.5 * np.linalg.norm(res)**2 + reg_term
 
     told = 1
     beta = 0.9
     prev_x = xhat.copy()
-    grad = np.asarray(At.dot(res))
+    grad = np.asarray(At.dot(confidence_array * res))
     qfval = prev_obj
 
     # Step size computation
```

Comment:
The computation of `L` (and` \mu`) can be changed to consider the `confidence_array`, @daducci do you know why is `L` computed like this?

Reply:
No, frankly no, I am not familiar with the FISTA code. Where did you find the details for modifying the code like this?

Reply:
I add the weights to the data term, as it is done in the WLS, and I did the computations, multiplications and gradient, to include them in FISTA.

Reply:
I checked the operations and, by doing a change of variable, I proposed a computation of `L` using the `confidence_array` that now is called `sqrt_W`